engine/runtime/src/main/java/org/enso/interpreter/runtime/warning/AppendWarningNode.java

```diff
@@ -0,0 +1,159 @@
+package org.enso.interpreter.runtime.warning;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Cached.Shared;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.InvalidArrayIndexException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.hash.EnsoHashMap;
+import org.enso.interpreter.runtime.data.hash.HashMapInsertNode;
+import org.enso.interpreter.runtime.data.vector.ArrayLikeAtNode;
+import org.enso.interpreter.runtime.data.vector.ArrayLikeLengthNode;
+
+@GenerateUncached
+public abstract class AppendWarningNode extends Node {
+
+  public static AppendWarningNode build() {
+    return AppendWarningNodeGen.create();
+  }
+
+  /**
+   * Appends a warning to the given object.
+   *
+   * @param object Object that will have the warning appended
+   * @param warnings Either an array-like object containing warnings to append, or a single warning.
+   *     It is expected that all the elements in the container are of {@link Warning} class.
+   * @return A wrapped object with warnings
+   */
+  public abstract WithWarnings execute(VirtualFrame frame, Object object, Object warnings);
```

Comment:
I'd give the method some more specific name. `executeAppend`?

Reply:
Done in 5003a916b708c2db0b3dfaea3f339df39ccf571c